augur/export_v2.py

```diff
@@ -45,6 +45,67 @@ def customformatwarning(message, category, filename, lineno, line=None):
 class InvalidOption(Exception):
     pass
 
+class CustomOrderedDict(OrderedDict):
+    """
+    Similar to OrderedDict but will convert dictionaries (and dictionaries of dictionaries)
+    into (nested) CustomOrderedDicts.
+    Encountered lists of dicts will be converted to lists of CustomOrderedDict but we will not
+    recursively explore nested lists.
+    Tuples and other iterators are not explored.
+    """
+    def __init__(self, *args):
+        super().__init__(*args)
+        for key in self:
+            if isinstance(self[key], dict) and not isinstance(self[key], OrderedDict):
+                self[key] = CustomOrderedDict(self[key])
+            elif isinstance(self[key], list):
+                self[key] = [
+                    (CustomOrderedDict(el) if (isinstance(el, dict) and not isinstance(el, OrderedDict)) else el)
+                    for el in self[key]
+                ]
+    def set_order(self, *order):
+        """
+        changes the order of keys to match those specified in `order` as much
+        as possible. Missing keys are ignored. Extra keys will come after those
+        specified in `order`.
+        """
+        for key in reversed(order):
+            self.move_to_end_if_present(key, last=False)
```

Comment:
I wish the `move_to_end(…, last = False)` variation was spelled `move_to_start(…)` :joy:

(I realize this is the stdlib's API, not yours!)

Reply:
Yeah, when I originally read that API I thought "why don't they have a move_to_start, that's what I want!?!" and only afterwards realised it was possible through the `last=False`.